test/gui/tst_sharing/test.feature

```diff
@@ -137,3 +137,34 @@ Feature: Sharing
         When the user creates a new public link for file "%client_sync_path%/textfile0.txt" with password "pass123" using the client-UI
         Then as user "Alice" the file "textfile0.txt" should have a public link on the server
         And the public should be able to download the file "textfile0.txt" with password "pass123" from the last created public link by "Alice" on the server
+
+
+    Scenario: user changes the expiration date of an already existing public link using webUI
+        Given user "Alice" has been created on the server with default attributes
+        And user "Alice" has set up a client with these settings and password "1234":
+            """
+            [Accounts]
+            0\Folders\1\ignoreHiddenFiles=true
+            0\Folders\1\localPath=%client_sync_path%
+            0\Folders\1\paused=false
+            0\Folders\1\targetPath=/
+            0\Folders\1\version=2
+            0\Folders\1\virtualFilesMode=off
+            0\dav_user=alice
+            0\display-name=Alice
+            0\http_oauth=false
+            0\http_user=alice
+            0\url=%local_server%
+            0\user=Alice
+            0\version=1
+            version=2
+            """
+        And user "Alice" has created a public link on the server with following settings
+            | path       | textfile0.txt |
+            | name       | Public link   |
+            | expireDate | 2038-10-14    |
+        When the user opens the public links dialog of "%client_sync_path%/textfile0.txt" using the client-UI
+        And the user edits the public link named "Public link" of file "textfile0.txt" changing following
+            | expireDate | 2038-07-21 |
+        Then the fields of the last public link share response of user "Alice" should include on the server
+            | expireDate | 2038-07-21 |
```

Comment:
that step fails for me with
```
AssertionError [ERR_ASSERTION]: Expiry Date Mismatch: [object Object]
    at Object.assertUserLastPublicShareDetails (/home/artur/www/owncloud-test-middleware/src/helpers/sharingHelper.js:431:16)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /home/artur/www/owncloud-test-middleware/src/app.js:56:7 {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: '2038-02-07',
  expected: '2038-07-21',
  operator: 'strictEqual'
}

Reply:
Did you run the test after removing the `snooze(5)` step?

Reply:
the problem seems to be my location settings. my date format is set to "d/m/yy" yours seems to be "m/d/yy"

Reply:
Will it be a problem in future if we run tests on CI?

Reply:
we need a way to set the env. variable `LC_TIME=en_US.utf8` before starting the app under test to make sure its always the same in all test runs

Reply:
I've fixed the branch, now the tests work for me. I had to set the env. variable in ` test/gui/envvars ` these are the variables that get passed on to the AUT see https://kb.froglogic.com/squish/howto/specifying-environment-variables-application/

Reply:
> the problem seems to be my location settings. my date format is set to "d/m/yy" yours seems to be "m/d/yy"

related to https://github.com/owncloud/client/issues/8367 ?

Reply:
no, my system is set to Australian date format, and I believe @swoichha is set to US, so the app behaives correctly, I just had to adjust the app under test